libraries/primitive_consul_health.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'primitive'
+require 'chef/mash'
+
+module Choregraphie
+  class ConsulHealthCheck < Primitive
+    def initialize(options = {})
+      @options = Mash.new(options)
+
+      @options[:tries] ||= 10
+      @options[:checkids] ||= []
+      @options[:delay] ||= 15
+
+      if @options[:consul_token]
+        require 'diplomat'
+        Diplomat.configure do |config|
+          config.acl_token = @options[:consul_token]
+        end
+      end
+
+      # this block could be used to configure diplomat if needed
+      yield if block_given?
+    end
+
+    def register(choregraphie)
+      choregraphie.cleanup do
+        require 'diplomat'
+        tries = @options[:tries]
+        all_passing = false
+        failing_checks = []
+
+        while !all_passing && tries != 0
```

Comment:
logic is easier to read but still complex, specifically with negations of all_passing/non_passing.

I suggest to drop all_passing variable:

```ruby
non_passing = [true]
while non_passing.any? && tries > 0 do
  checks = Diplomat::Check.checks
  non_passing = ...
  non_passing.each { ... }
  if non_passing.any?
    # log
    tries -= 1
    sleep ...
  end
end
```

but a better and more rubyway would be:
```
tries.times do |i|
  # definition of non_passing
  ...
  break if non_passing.empty?
  # log ...
  sleep ...
end
```

you can also remove the failing_checks variable which does not seem used at the moment.